substrate/client/offchain/src/api/http.rs

```diff
@@ -48,21 +48,26 @@ use std::{
 
 const LOG_TARGET: &str = "offchain-worker::http";
 
+pub type Body = BoxBody<hyper::body::Bytes, hyper::Error>;
+
+type Sender = mpsc::Sender<Result<hyper::body::Frame<hyper::body::Bytes>, hyper::Error>>;
+type Receiver = mpsc::Receiver<Result<hyper::body::Frame<hyper::body::Bytes>, hyper::Error>>;
+
+type HyperClient = client::Client<HttpsConnector<client::connect::HttpConnector>, Body>;
+type LazyClient = Lazy<HyperClient, Box<dyn FnOnce() -> HyperClient + Send>>;
+
 /// Wrapper struct used for keeping the hyper_rustls client running.
 #[derive(Clone)]
-pub struct SharedClient(Arc<Lazy<HyperClient<HttpsConnector<client::HttpConnector>, Body>>>);
+pub struct SharedClient(Arc<LazyClient>);
 
 impl SharedClient {
-	pub fn new() -> Self {
-		Self(Arc::new(Lazy::new(|| {
-			let connector = HttpsConnectorBuilder::new()
-				.with_native_roots()
-				.https_or_http()
-				.enable_http1()
-				.enable_http2()
-				.build();
-			HyperClient::builder().build(connector)
-		})))
+	pub fn new() -> std::io::Result<Self> {
+		let builder = HttpsConnectorBuilder::new()
+			.with_provider_and_native_roots(rustls::crypto::ring::default_provider())?;
+		Ok(Self(Arc::new(Lazy::new(Box::new(|| {
+			let connector = builder.https_or_http().enable_http1().enable_http2().build();
+			client::Client::builder(TokioExecutor::new()).build(connector)
+		})))))
```

Comment:
The above boxing is needed to resolve this; This should be a `Lazy` with closure, otherwise we cannot return `std::io::Result<Self>` and `Self.0` becomes `Lazy<std::io::Result<HyperClient>>`, which is quite inconvenient to use